docs/the-new-architecture/pure-cxx-modules.md

```diff
@@ -318,46 +318,95 @@ If you did everything right, your project on the left should look like this:
+2. Select the `Cocoa Touch Class` template
+3. Add the name `SampleNativeModuleProvider` (keep the other field as `Subclass of: NSObject` and `Language: Objective-C`)
+4. Click Next to generate the files.
+5. Rename the `SampleNativeModuleProvider.m` to `SampleNativeModuleProvider.mm`. The `mm` extension denotes an Objective-C++ file.
+6. Implement the content of the `SampleNativeModuleProvider.h` with the following:
 
-```diff title="SampleApp/AppDelegate.mm"
-#import <React/RCTBundleURLProvider.h>
-+ #import <RCTAppDelegate+Protected.h>
-+ #import "NativeSampleModule.h"
+```objc title="NativeSampleModuleProvider.h"
 
-// ...
-  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
-#endif
-}
+#import <Foundation/Foundation.h>
+#import <ReactCommon/RCTTurboModule.h>
 
-+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
-+                                                      jsInvoker:(std::shared_ptr<facebook::react::CallInvoker>)jsInvoker
-+{
-+  if (name == "NativeSampleModule") {
-+    return std::make_shared<facebook::react::NativeSampleModule>(jsInvoker);
-+  }
-+
-+  return [super getTurboModule:name jsInvoker:jsInvoker];
-+}
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NativeSampleModuleProvider : NSObject <RCTModuleProvider>
 
 @end
+
+NS_ASSUME_NONNULL_END
 ```
 
-These changes are doing a few things:
+This declares a `NativeSampleModuleProvider` object that conforms to the `RCTModuleProvider` protocol.
 
-1. Importing the `RCTAppDelegate+Protected` header to make it visible to the AppDelegate that it is conforming to the `RCTTurboModuleManagerDelegate` protocol.
-2. Importing the Pure C++ Native Turbo Module interface `NativeSampleModule.h`
-3. Overriding the `getTurboModule` method for C++ modules so that when the JS side asks for a module called `NativeSampleModule`, the app knows which module has to be returned.
+7. Implement the content of the `SampleNativeModuleProvider.mm` with the following:
+
+```objc title="NativeSampleModuleProvider.mm"
+
+#import "NativeSampleModuleProvider.h"
+#import <ReactCommon/CallInvoker.h>
+#import <ReactCommon/TurboModule.h>
+#import "NativeSampleModule.h"
+
+@implementation NativeSampleModuleProvider
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+  return std::make_shared<facebook::react::NativeSampleModule>(params.jsInvoker);
+}
+
+@end
+```
+
+This code implements the `RCTModuleProvider` protocol by creating the pure C++ `NativeSampleModule` when the `getTurboModule:` method is called.
+
+##### 3.2 Update the package.json
+
+The last step consist in updating the `package.json` to tell React Native about the link between the JS specs of the Native Module and the concrete implementation of those spec in native code.
+
+Modify the `package.json` as it follows:
+
+```json title="package.json"
```

Comment:
nit: should this probably be a `diff` because you're suggesting a modification to an existing file?

Reply:
it is a diff. There are the #highlight-add-start / #highlight-end-start annotation that will highlights the changes, while keeping the json syntax highlight.